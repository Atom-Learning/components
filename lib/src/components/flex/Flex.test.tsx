import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { Flex } from '.'

const FlexImplementation = (props) => {
  return <Flex align="center" justify="center" {...props} />
}

describe(`Flex component`, () => {
  it('renders', async () => {
    const { container } = render(<FlexImplementation>Flex</FlexImplementation>)
    await screen.findByText('Flex')
    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    render(<FlexImplementation />)

    const results = await axe(document.body)
    expect(results).toHaveNoViolations()
  })
})
