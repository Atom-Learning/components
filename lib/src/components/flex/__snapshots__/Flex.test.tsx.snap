// Vitest Snapshot v1

exports[`Flex component > renders 1`] = `
<div>
  <div
    class="c-dhzjXW c-dhzjXW-ikLAKdM-css"
  >
    FLEX
  </div>
</div>
`;
